axes/handlers/database.py

```diff
@@ -119,58 +119,48 @@ def user_login_failed(
         # 2. database query: Calculate the current maximum failure number from the existing attempts
         failures_since_start = 1 + self.get_failures(request, credentials)
 
-        # 3. database query: Insert or update access records with the new failure data
-        try:
-            attempt = AccessAttempt.objects.get(
+        if not (settings.AXES_ONLY_USER_FAILURES and username is None):
```

Comment:
Doesn't this alter the logic of the code by skipping attempt fetching in case where username is not set? The old logic fetches an access attempt with `username = None` existing in the database and this logic skips that case, right? I'm not sure if that behaviour is of sound design, but this is a change to it nonetheless, and imo should be considered.

Also, I think this logic could be reverted to run the check + logging first so that we immediately return from the function in case the correct conditions are not met, similar to https://github.com/jazzband/django-axes/blob/84b5ccd5d397694d6b04e72e84de9c5861183294/axes/handlers/database.py#L94...L98

Reply:
In the previous implementation, the ```user=None``` case would never have create a database entry, which could have been found in step 3 (prevented due to the if statement in the fallback for new entries), so i needed to pull the condition up to prevent in this case the creation of the entry for no user information.

Reply:
Yeah, I agree it didn't do anything sane.

I'd still vouch for reverting this condition to something like:

```python
if settings.AXES_ONLY_USER_FAILURES and username is None:
    log.warning(
        "AXES: Username is None and AXES_ONLY_USER_FAILURES is enabled. New record will NOT be created."
    )
    return
```